docs/guides/templates.md

```diff
@@ -0,0 +1,41 @@
+# Templates
+
+When a Function is created, an example implementation and a Function metadata file are written into the new Function's working directory.  Together, these files are referred to as the Function's Template.  Included are the templates 'http' and 'events' for each supported language runtime.
+
+These embedded templates are minimal by design.  The Function contains a minimum of external dependencies, and the 'func.yaml' defines a final environment within which the Funciton will execute that is devoid of any extraneous packages or services.
+
+To make use of more complex inital Function implementions, or to define runtime environments with arbitrarily complex requirements, the templates system is fully pluggable.
+
+## External Git Repositories
+
+When creating a new Function, a Git repository can be specified as the source for the template files.  For example, the Boson Project maintains a set of example Functions at https://github.com/boson-project/templates which can be used during project creation.
+
+For example, the Boson Project Templates repository contains an example "Hello World" Function implementation in each of the officially supported languages.  To use this template via the CLI, use the flags:
+
+func create <name> --template hello-world --repository https://github.com/boson-project/templates
```

Comment:
Wouldn't this need the language specified as well? E.g.
```suggestion
func create <name> -l ruby --template hello-world --repository https://github.com/boson-project/templates
```
And it might be helpful to show what the repository's directory structure looks like.
```
├── README.md
└── ruby
    └── hello-world
        ├── func.rb
        └── Gemfile
```

Reply:
I left out the language because it has a default, and soon with `config --global` a user will be able to set a global default for their primary language.  Furthermore, there is a placeholder function in each language, so it won't matter which they choose.  Do you think this looks like an error or unintentional omission?  If so, I will commit your suggestion.

Reply:
The reason I left out any text about the internal structure of a repository is beacuse this document is from the perspective of the user of the repository, who may be better served by focusing just on the command structure.  A forthcoming document specifically targed at creating custom/external repositories would have this information in detail, however.  Do you still think this section would be improved by adding the directory structure?  Or perhaps should I add a link here: "Read more about custom repositories in [Creating a Repository](custom_repositories.md)"  and include in that document a full explanation?

Reply:
>  I left out the language because it has a default

What is the default language for remote repositories? And how does the user know what that default is?

> The reason I left out any text about the internal structure of a repository is because this document is from the perspective of the user of the repository, who may be better served by focusing on how to choose a template from a repository.

OK I see your point - it does make sense from that perspective to keep the commands simple. But maybe mentioning that there is a default, and what that default is would also be useful information?

> Or perhaps I should go all the way and create that document now, and add a link here such as: "Read more about how repositories are structured in [Creating a Repository](custom_repositories.md)"?

This would be great. From the upstream/task force perspective, I think it's important to really highlight how easy it is for vendors/providers to create, publish and support these repositories. So a document around that would really help this PR, in my opinion.

Reply:
You bet.  Working on it now and will drop a link here.